### PR TITLE
MFLOW-29: Implement simple alerting utility to replace console.error with Compass alerting

### PR DESCRIPTION
**Pull request by b1scuit**
## Summary

This PR implements a simple alerting utility that serves as a drop-in replacement for `console.error()` calls throughout the MFlow codebase, forwarding alerts to Compass for centralized monitoring.

**Key Features:**
- ✅ **Frontend Utility**: `compassAlert()` function in `src/utils/compassAlert.ts`
- ✅ **Backend Cloud Function**: `compassAlert` function in `functions/src/index.ts`
- ✅ **Graceful Fallback**: Still logs to console when cloud function is unavailable
- ✅ **Error Handling**: Comprehensive retry logic with exponential backoff
- ✅ **Helper Functions**: `compassAlertError()` and `compassAlertPerformance()` utilities
- ✅ **100% Test Coverage**: Complete test suites for both frontend and backend

## Usage

```typescript
// Before
console.error('Failed to process ticket', { ticketId, error });

// After
compassAlert('Failed to process ticket', { ticketId, error });
```

### Frontend (`src/utils/compassAlert.ts`)
- Drop-in replacement for `console.error()` 
- Logs to console immediately for local debugging
- Sends alerts to Firebase cloud function asynchronously
- Extracts source location from stack traces
- Handles Firebase Functions integration gracefully

### Backend (`functions/src/index.ts`)
- Firebase Functions v2 with `onCall` pattern
- Authenticates with Compass API using environment variables
- Implements exponential backoff retry logic (1s, 2s, 4s)
- Handles rate limiting with retry-after header support
- Comprehensive error handling and logging

### Environment Variables Required
- `COMPASS_API_URL`: Compass API endpoint
- `COMPASS_API_KEY`: Authentication key for Compass

## Test Results

- **Frontend**: 20 tests passing with 100% coverage
- **Backend**: 37 tests passing with 100% coverage
- **Integration**: All existing tests continue to pass
- **Build**: TypeScript compilation and production build successful

## Test Plan

- [x] Unit tests for frontend utility function
- [x] Unit tests for backend cloud function
- [x] Integration tests with Firebase Functions
- [x] Error handling and retry logic verification
- [x] Performance and edge case testing
- [x] Existing test suite regression testing
- [x] TypeScript type checking
- [x] Production build verification

## Deployment Notes

1. **Environment Variables**: Set `COMPASS_API_URL` and `COMPASS_API_KEY` in Firebase Functions environment
2. **Cloud Function**: Deploy updated functions with `cd functions && npm run deploy`
3. **Usage**: Import and use throughout codebase as needed

## Links

- **Jira Ticket**: [MFLOW-29](https://debuggin-ltd.atlassian.net/browse/MFLOW-29)
- **Branch**: `MFLOW-29-compass-alerting-utility`

🤖 Generated with [Claude Code](https://claude.ai/code)

[MFLOW-29]: https://debuggin-ltd.atlassian.net/browse/MFLOW-29?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: functions/src/index.ts [hunk 1]
```diff
@@ -79,3 +79,135 @@ export const tokenExchange = onCall({
...
+        const { message, context, timestamp, source } = request.data;
+
+        // Validate required fields
+        if (!message) {
+            throw new HttpsError('invalid-argument', 'Alert message is required');
+        }
+
+        // Get Compass API credentials from environment variables
+        const compassApiUrl = process.env.COMPASS_API_URL;
+        const compassApiKey = process.env.COMPASS_API_KEY;
```
Comment (b1scuit) [inline functions/src/index.ts]: Can this use the secrets that exist instead, in the format 

(atlassian-email):(atlassian-api-key) as a basic auth

File: functions/src/index.ts [hunk 2]
```diff
@@ -79,3 +79,138 @@ export const tokenExchange = onCall({
...
+        const { message, context, timestamp, source } = request.data;
+
+        // Validate required fields
+        if (!message) {
+            throw new HttpsError('invalid-argument', 'Alert message is required');
+        }
+
+        // Get Compass API credentials from environment variables
+        const compassApiUrl = process.env.COMPASS_API_URL;
+        const atlassianCredentials = process.env.ATLASSIAN_CREDENTIALS;
```
Comment (b1scuit) [inline functions/src/index.ts]: Pull the secret in from GCP Secrets manager

Comment (b1scuit) [reply functions/src/index.ts]: the secrets atlassian-email and atlassian-api-key
